File: 2025_03/프로그래머스 - Lv.1/약수의 개수와 덧셈/약수의 개수와 덧셈.js
// 제곱수는 약수가 홀수개임
function solution(left, right) {
  var answer = 0;
  for(let i = left ; i <= right ; i++){
    answer += parseInt(i**0.5) === i**0.5 ? -i : i
  }
  return answer;
}

File: 2025_03/프로그래머스 - Lv.1/약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    for i in range(left, right+1):
        cnt = 0
        for j in range(1, i+1):
            cnt += 1 if i % j == 0 else 0
        answer += i if cnt % 2 == 0 else -i
    return answer


# 제곱수 => 약수의 개수가 홀수개
def solution(left, right):
    answer = 0
    for i in range(left,right+1):
        answer += -i if int(i**0.5) == i**0.5 else i
    return answer